lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  await migratePdrRecord(testPdr, knex);
+
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
+      status: testPdr.status,
+      progress: testPdr.progress,
+      pan_sent: testPdr.PANSent,
+      pan_message: testPdr.PANmessage,
+      stats: testPdr.stats,
+      address: testPdr.address,
+      original_url: testPdr.originalUrl,
+      timestamp: new Date(testPdr.timestamp),
+      duration: testPdr.duration,
+      created_at: new Date(testPdr.createdAt),
+      updated_at: new Date(testPdr.updatedAt),
+    },
+    ['updated_at'])
+  );
+});
+
+test.serial('migratePdrRecord throws SchemaValidationError on invalid source data from DynamoDB', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+
+  delete testPdr.status;
+
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { name: 'SchemaValidationError' }
+  );
+});
+
+test.serial('migratePdrRecord handles nullable fields on source PDR data', async (t) => {
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+
+  delete testPdr.execution;
+  delete testPdr.PANSent;
+  delete testPdr.PANmessage;
+  delete testPdr.progress;
+  delete testPdr.stats;
+  delete testPdr.address;
+  delete testPdr.originalUrl;
+  delete testPdr.timestamp;
+  delete testPdr.duration;
+  delete testPdr.updatedAt;
+
+  await migratePdrRecord(testPdr, knex);
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      ...testPdr,
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
+      status: testPdr.status,
+      progress: null,
+      pan_sent: null,
+      pan_message: null,
+      stats: null,
+      address: null,
+      original_url: null,
+      timestamp: null,
+      duration: null,
+      created_at: new Date(testPdr.createdAt),
+      updated_at: new Date(testPdr.updatedAt),
+    }, ['createdAt', 'updatedAt', 'pdrName', 'collection', 'provider', 'collectionId', 'updated_at'])
```

Comment:
I guess it's debatable, but the use of `omit` here seems more complicated than just removing the `...testPdr` on line 213 and instead constructing the exact expected object with only the expected properties

Reply:
It actually wasn't that complicated to remove `...testPdr` and construct the exact expected object. Only thing was `updatedAt` is a nullable field, but I think, upon insertion, `updated_at` will be set. So I need to remove it from the comparison.